src/main/java/au/org/emii/ncdfgenerator/NcdfEncoderBuilder.java

```diff
@@ -0,0 +1,1476 @@
+		}
+		return null;
+	}
+
+	private ExprLiteral parseLiteral( String s, int pos )
+	{
+		// TODO pos2
+		int pos2 = pos;
+		// ignore escaping for the moment
+		if(s.charAt(pos2) != '\'')
+			return null;
+		++pos2;
+
+		while(s.charAt(pos2) != '\'')
+			++pos2;
+
+		++pos2;
+		return new ExprLiteral(pos, s.substring( pos + 1, pos2 - 1));
+	}
+}
+
+
+class PrettyPrinterVisitor implements IExprVisitor
+{
+
+	PrettyPrinterVisitor( OutputStream os)
+	{
+		this.os = os;
+	}
+
+	OutputStream	os;
+
+	private void write( String s )
+		throws IOException
+	{
+		os.write( s.getBytes(Charset.forName("UTF-8")));
+	}
+
+	public void visit( ExprSymbol expr ) throws Exception
+	{
+		write( "Symbol:" + expr.value );
+	}
+
+	public void visit(  ExprInteger expr ) throws Exception
+	{
+		write( "Integer:" + expr.value );
+	}
+
+	public void visit( ExprTimestamp expr ) throws Exception
+	{
+		write( "Timestamp:" + expr.value );
+	}
+
+	public void visit(  ExprLiteral expr ) throws Exception
+	{
+		write( "Literal:" + expr.value );
+	}
+
+	public void visit( ExprProc expr ) throws Exception
+	{
+		write( "(" + expr.symbol + " " );
+		for( IExpression child : expr.children ) {
+			child.accept(this);
+			write(  " ");
+		}
+		write( ")" );
+	}
+}
+
+
+class PGDialectSelectionGenerator implements IExprVisitor
+{
+	StringBuilder b;
+	SimpleDateFormat df;
+
+
+	public PGDialectSelectionGenerator( StringBuilder b )
+	{
+		this.b = b;
+		this.df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+	}
+
+	// think our naming is incorrect
+	public void visit( ExprInteger expr )
+	{
+		// This should actually emit a '?' and load the value into the sql parameter list
+		// to avoid sql injection
+		b.append( expr.value );
+	}
+
+	public void visit( ExprTimestamp expr )
+	{
+		b.append( "'" + df.format(expr.value ) + "'" );
+	}
+
+	public void visit( ExprLiteral expr )
+	{
+		b.append("'"+ expr.value + "'");
+	}
+
+	public void visit( ExprSymbol expr )
+	{
+		// must quote field to enforce full case handling
+		b.append('\"' + expr.value + '\"' );
+	}
+
+	public void visit( ExprProc expr ) throws Exception
+	{
+		String symbol = expr.symbol;
+
+		if(symbol.equals("equals")) {
+			emitInfixSqlExpr( "=", expr );
+		}
+		else if(symbol.equals("gt")) {
+			emitInfixSqlExpr( ">", expr );
+		}
+		else if(symbol.equals("lt")) {
+			emitInfixSqlExpr( "<", expr );
+		}
+		else if(symbol.equals("geq")) {
+			emitInfixSqlExpr( ">=", expr );
+		}
+		else if( symbol.equals("and")
+			|| symbol.equals("or")
+			) {
+			emitInfixSqlExpr( symbol, expr );
+		}
+		else {
+			throw new NcdfGeneratorException( "Unrecognized proc expression symbol '" + symbol + "'" );
+		}
+	}
+
+	public void emitInfixSqlExpr( String op, ExprProc expr ) throws Exception
+	{
+		// if expansion is done in order we may be ok,....
+		b.append('(');
+		expr.children.get(0).accept(this);
+		b.append(' ');
+		b.append(op);
+		b.append(' ');
+		expr.children.get(1).accept(this);
+		b.append(')');
+	}
+}
+
```

Comment:
There's a lot of code to parse the subset parameters here.  Using geotools cql parser may be better?